crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -182,14 +183,16 @@ pub(super) enum TimelineEventKind {
 
 impl TimelineEventKind {
     /// Creates a new `TimelineEventKind` with the given event and room version.
-    pub fn from_event(
+    pub async fn from_event<P: RoomDataProvider>(
         event: AnySyncTimelineEvent,
-        room_version: &RoomVersionId,
+        raw_event: &Raw<AnySyncTimelineEvent>,
```

Comment:
ugh, having to pass both `event` and `raw_event` is a bit ugly, but it might have to do for now. Let's at least document them though (ie, that `event` is the deserialized `raw_event`)

Reply:
Added some documentation:
```
        // We need also the [`Raw`] version of the `event` because unknown json fields are lost
        // when de-serialising to the actual `AnySyncTimelineEvent` type. And we need some
        // of these field to determine the `UtdCause`.
 ```
 
 I wonder if we couldn't pass both of them to `UtdCause::Determine`? Would avoid a deserialisation there, or maybe we could raw access the `origin_server_ts`